tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Auth.java

```diff
@@ -0,0 +1,191 @@
+package org.unicode.cldr.web.api;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
+import org.unicode.cldr.web.CookieSession;
+import org.unicode.cldr.web.UserRegistry;
+import org.unicode.cldr.web.UserRegistry.LogoutException;
+import org.unicode.cldr.web.UserRegistry.User;
+import org.unicode.cldr.web.WebContext;
+
+@Path("/auth")
+public class Auth {
+
+    private final boolean DEBUG = false;
+
+    @Path("/login")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Operation(
+        summary = "Login and get a CookieSession id",
+        description = "This logs in with a username/password and returns a CookieSession")
+    @APIResponses(
+        value = {
+            @APIResponse(
+                responseCode = "200",
+                description = "CookieSession",
+                content = @Content(mediaType = "application/json",
+                    schema = @Schema(implementation = LoginResponse.class))),
+            @APIResponse(
+                responseCode = "401",
+                description = "Authorization required"),
+            @APIResponse(
+                responseCode = "403",
+                description = "Forbidden, account may be locked"),
+        })
+    public Response login(
+        @Context HttpServletRequest hreq,
+        // @QueryParam("ss") @Schema(required = true, description = "Session String") String sessionString,
+        LoginRequest request) {
+        if (request.isEmpty()) {
+            return Response.status(401, "Authorization required").build();
+        }
+
+        final String ip = WebContext.userIP(hreq);
+
+        try {
+            LoginResponse resp = new LoginResponse();
+
+            // we start with the user
+            User user = CookieSession.sm.reg.get(request.password, request.email, ip, false);
+            if(user == null) {
+                return Response.status(403, "Login failed").build();
+            } if(UserRegistry.userIsLocked(user)) {
+                return Response.status(403, "User Locked").build();
```

Comment:
formatting

Reply:
need to find out how to run the eclipse formatter from the cli

Reply:
for js I'm happy with npx prettier -- there is a java plugin, but I haven't tried it: https://github.com/jhipster/prettier-java

btw what are you using on vue files?

Reply:
cleaned up